Feature sign in

christiangelone opened the pull request:
## Summary

 * Adds endpoint for session token retrieval by creds.
 * Adds tests for happy path, invalid email & invalid password.

## Known Issues

 * No issues

## Trello Card
https://trello.com/c/1q7Wpv2b

Commit:
Adds endpoint with its happy path test for session token

Commit:
Covers not to return a token when email is invalid

Commit:
Covers not to return a token when password is invalid

Commit:
Merge branch 'master' of https://github.com/wolox-training/cha-express-js into feature_sign-in

Commit:
Adds logging to user session endpoint

gonziesc commented on app/controllers/users.js:
```diff
@@ -4,8 +4,47 @@ const bcrypt = require('bcryptjs');
+    if (user !== null) {
+      return bcrypt.compare(creds.password, user.password);
+    }
+    throw new Error('invalid email');
```
promise reject here

gonziesc commented on app/controllers/users.js:
```diff
@@ -4,8 +4,47 @@ const bcrypt = require('bcryptjs');
+    throw new Error('invalid email');
+  });
+
+  Promise.all([userPromise, matchPromise])
```
theres no need of promise all, just chain the promises

gonziesc commented on app/controllers/users.js:
```diff
@@ -4,8 +4,47 @@ const bcrypt = require('bcryptjs');
+          id: user.id
+        });
+      }
+      throw new Error('invalid password');
```
promise reject

gonziesc commented on app/middlewares/errors.js:
```diff
@@ -5,7 +5,7 @@ const DEFAULT_STATUS_CODE = 500;
 const statusCodes = {
   [errors.BAD_REQUEST]: 400,
-  [errors.INVALID_USER]: 401,
+  [errors.INVALID_CREDS]: 401,
```
CREDENTIALS

gonziesc commented on app/services/jwt.js:
```diff
@@ -0,0 +1,29 @@
+
+const SECRET = 'secret'; // should go on env variable
+
+exports.HEADER_NAME = 'auth_token'; // should go on env variable
```
resolve this comments

Commit:
Removes promise.all from session endpoint
Makes invalid credentials error more cleaner
Replaces throw error with Promise.reject
Puts auth header & jwt secret in env variables

gonziesc commented on app/controllers/users.js:
```diff
@@ -4,8 +4,50 @@ const bcrypt = require('bcryptjs');
+            next(errors.invalidCredentials(err));
+          });
+      } else {
+        return Promise.reject(new Error('invalid email'));
```
use next here

gonziesc reviewed (changes requested):
just that comment and fix ci

Commit:
Replace Promise.reject with next(...)

Commit:
Adds jwt seret & auth header for testing environment

gonziesc commented on app/controllers/users.js:
```diff
@@ -4,8 +4,44 @@ const bcrypt = require('bcryptjs');
+              id: user.id
+            });
+          }
+          return Promise.reject(new Error('invalid password'));
```
next here

gonziesc commented on test/controllers/user.test.js:
```diff
@@ -9,206 +9,311 @@ const server = require('../../app');
+            password: 'johndoeland'
+          };
+
+          return request.post('/users/sessions').send(userCreds);
```
you can delete the ; and just chain everything with the then

gonziesc commented on test/controllers/user.test.js:
```diff
@@ -9,206 +9,311 @@ const server = require('../../app');
+          done(new Error('Successful response - This should not be called'));
+        })
+        .catch(err => {
+          console.log(JSON.stringify(err.response.body, null, 2));
```
delelte console log

gonziesc commented on test/controllers/user.test.js:
```diff
@@ -9,206 +9,311 @@ const server = require('../../app');
+            password: 'johndoeinvalid'
+          };
+
+          return request.post('/users/sessions').send(userCreds);
```
same comment

Commit:
Attends comments over pull request: deletes console.log & chains promises

gonziesc commented on app/controllers/users.js:
```diff
@@ -4,8 +4,44 @@ const bcrypt = require('bcryptjs');
+              id: user.id
+            });
+          }
+          next(errors.invalidCredentials(new Error('invalid password')));
```
i'd do if(!match) next ... to make it more clear

gonziesc commented on test/controllers/user.test.js:
```diff
@@ -9,206 +9,316 @@ const server = require('../../app');
+        password: 'johndoereloaded'
+      };
+
+      request
```
return and chain the promises

gonziesc commented on test/controllers/user.test.js:
```diff
@@ -9,206 +9,316 @@ const server = require('../../app');
+        password: 'johndoereloaded'
+      };
+
+      request
```
same comment

gonziesc commented on test/controllers/user.test.js:
```diff
@@ -9,206 +9,316 @@ const server = require('../../app');
+            password: 'johndoeland'
+          };
+
+          return request
```
same

gonziesc commented on test/controllers/user.test.js:
```diff
@@ -9,206 +9,316 @@ const server = require('../../app');
+            password: 'johndoeinvalid'
+          };
+
+          return request
```
same

Commit:
Places (if not match) as suggested, and in tests it returns and chain promises

christiangelone committed:
Merge branch 'master' into feature_sign-in
